### DISCUSSION TITLE
How can the service_role key be renewed?

### DISCUSSION THREAD

**Discussion by dostuffthatmatters** [supabase/supabase#2265] [Questions] [answered]
For security reasons I want to revoke the previously distributed service_role key ... and create a new one. Is that possible without creating a new project altogether?

**Comment by silentworks** [comment]
You can do so by making a request from the dashboard feedback widget for the Supabase team to do so or you can email support@supabase.io to request this.

**Reply by dostuffthatmatters** [reply]
Ok, thank you! There should be an easier way in the future.

**Comment by soedirgo** [comment, marked as answer]
The `service_role` key is just a JWT, so you can mint new ones with the provided JWT secret. Unfortunately I don't think you can revoke JWTs, so if you *really* have to revoke it (it got leaked, etc.) you might be better off migrating to a new project. (Team members who are more familiar with auth can CMIIW.)

**Reply by steve-chavez** [reply]
> Unfortunately I don't think you can revoke JWTs

It could be done at the db level, like mentioned here: https://postgrest.org/en/v8.0/tutorials/tut1.html#bonus-topic-immediate-revocation

It would require including the [pre-request](https://postgrest.org/en/v8.0/configuration.html#pre-request) config on the Supabase Admin UI.
